indico/modules/rb_new/controllers/backend/bookings.py

```diff
@@ -153,3 +158,57 @@ def _process(self):
         booking_details['occurrences'] = occurrences
         booking_details['date_range'] = date_range
         return jsonify(booking_details)
+
+
+class RHBookingStateActions(RHRoomBookingBase):
+    def _process_args(self):
+        self.booking = Reservation.get_one(request.view_args['booking_id'])
+        self.action = request.view_args['action']
+
+    def _check_access(self):
+        RHRoomBookingBase._check_access(self)
+        if self.action == 'approve':
```

Comment:
This could be simplified like this:

```
funcs = {'approve': self.booking.can_be_accepted,
         'reject': self.booking.can_be_rejected,
         'cancel': self.booking.can_be_cancelled}
if not funcs[self.action](session.user):
    raise Forbidden
```

(or simply merge the inner ifs with the outer ones)